ecommerce_integrations/woocommerce/woocommerce_connection.py

```diff
@@ -172,27 +170,32 @@ def rename_address(address, customer):
 def link_items(items_list, woocommerce_settings, sys_lang):
 	for item_data in items_list:
 		item_woo_com_id = cstr(item_data.get("product_id"))
-
-		if not frappe.db.get_value("Ecommerce Item", {"erpnext_item_code": item_woo_com_id}, "name"):
+		tmp_erpnext_item_code = _("woocommerce - {0}", sys_lang).format(item_woo_com_id)
+		if not frappe.db.get_value(
+			"Ecommerce Item", {"erpnext_item_code": tmp_erpnext_item_code}, "name"
+		):
+			if not frappe.db.get_value("Item", {"woocommerce_id": item_woo_com_id}, "name"):
+				create_erp_item(
+					item_woo_com_id, woocommerce_settings, item_data, sys_lang, tmp_erpnext_item_code
+				)
 			item = frappe.new_doc("Ecommerce Item")
 			item.integration = MODULE_NAME
-			if not frappe.db.get_value("Item", {"woocommerce_id": item_woo_com_id}, "name"):
-				create_erp_item(item_woo_com_id, woocommerce_settings, item_data, sys_lang)
-			item.erpnext_item_code = item_woo_com_id
+			item.erpnext_item_code = tmp_erpnext_item_code
 			item.integration_item_code = item_data.get("sku")
-			item.has_variants: 0  # to check
+			item.has_variants = 0  # to check
 			item.sku = item_data.get("sku")
 			item.flags.ignore_mandatory = True
 			item.save(ignore_permissions=True)
 
 
-def create_erp_item(item_woo_com_id, woocommerce_settings, item_data, sys_lang):
+def create_erp_item(item_woo_com_id, woocommerce_settings, item_data, sys_lang, erpnext_item_code):
 	item = frappe.new_doc("Item")
-	item.item_code = _("woocommerce - {0}", sys_lang).format(item_woo_com_id)
+	item.item_code = erpnext_item_code
 	item.stock_uom = woocommerce_settings.uom or _("Nos", sys_lang)
 	item.item_group = _("WooCommerce Products", sys_lang)
 
 	item.item_name = item_data.get("name")
+	item.sku = item_data.get("sku")
```

Comment:
Don't feel this is necessary if we are using e-commerce item for mapping.

Reply:
Will refactor to avoid it